common/net/bolt1.py

```diff
@@ -0,0 +1,283 @@
+
+        payload = num_pong_bytes.to_bytes(2, "big")
+
+        if ignored_bytes:
+            if not isinstance(ignored_bytes, bytes):
+                raise TypeError("ignored_bytes must be bytes if set")
+            if len(ignored_bytes) > pow(2, 16) - 4:
+                raise ValueError(f"ignored_bytes cannot be higher than {pow(2, 16) - 4}")
+            payload += len(ignored_bytes).to_bytes(2, "big") + ignored_bytes
+
+        super().__init__(message_types["ping"], payload)
+        self.num_pong_bytes = num_pong_bytes
+        self.ignored_bytes = ignored_bytes
+
+    @classmethod
+    def from_bytes(cls, message):
+        """Builds a PingMessage from its byte representation."""
+        # Message should be at least: type (2-byte) + num_pong_bytes (2-bytes) + byteslen (2-bytes)
+        message_sanity_checks(message, message_types["ping"], 6)
+        num_pong_bytes = int.from_bytes(message[2:4], "big")
+        byteslen = int.from_bytes(message[4:6], "big")
+
+        if byteslen:
+            ignored = message[6 : 6 + byteslen]
+            if len(ignored) < byteslen:
+                raise ValueError("Wrong message format. Unexpected EOF")
+            if len(message) > 6 + byteslen:
+                raise ValueError("Wrong data format. message has additional tailing data")
+            return cls(num_pong_bytes, ignored)
+
+        return cls(num_pong_bytes)
+
+
+class PongMessage(Message):
+    """
+    Message to be sent in response to a ``PingMessage``.
+
+    Args:
+        ignored_bytes (:obj:`bytes`): filling bytes added to the message by the sender. Should match the ones requested
+            by the sender of the ``PingMessage``.
+    """
+
+    def __init__(self, ignored_bytes=None):
+        if ignored_bytes:
+            if not isinstance(ignored_bytes, bytes):
+                raise TypeError("ignored_bytes must be bytes if set")
+            if len(ignored_bytes) > pow(2, 16) - 4:
+                raise ValueError(f"ignored_bytes cannot be higher than {pow(2, 16) -4}")
+
+            payload = len(ignored_bytes).to_bytes(2, "big") + ignored_bytes
+
+        else:
+            payload = int.to_bytes(0, 2, "big")
+
+        super().__init__(message_types["pong"], payload)
+        self.ignored_bytes = ignored_bytes
+
+    @classmethod
+    def from_bytes(cls, message):
+        """Builds a PongMessage from its byte representation."""
+        # Message should be at least: type (2-byte) + byteslen (2-bytes)
+        message_sanity_checks(message, message_types["pong"], 4)
+        byteslen = int.from_bytes(message[2:4], "big")
+
+        if byteslen:
+            ignored_bytes = message[4 : 4 + byteslen]
+            if len(ignored_bytes) < byteslen:
+                raise ValueError("Wrong message format. Unexpected EOF")
+            if len(message) != 4 + byteslen:
+                raise ValueError("Wrong data format. message has additional tailing data")
```

Comment:
as the others, typo "trailing" and capital M.

Reply:
FIt in d744359630eae32899372242661994ca1ef294d7

Reply:
Fixed in d744359630eae32899372242661994ca1ef294d7